### PR TITLE
SBERDOMA-704 Export tickets to excel review fixes

### PR DESCRIPTION
**Pull request by dkoviazin**
This PR is on Apollo https://apollo.dev.doma.ai/ - can be tested there
 
1. Locale dependant words do not present in the export file
2. All dates are converted to timezone for the user who makes the export file
3. Moved access check from query body to access check (@pahaz)
4. Added ticket comments to export file
5. Dates of status changes are now from the ticket change history
6. Fix generate tickets util

P.S Export file better to open in MS Office or Google Docs

[exportTicketsExample.xlsx](https://github.com/open-condo-software/condo/files/6814239/exportTicketsExample.xlsx)

### CODE REVIEW

**Review by sitozzz — APPROVED** [review]

**Comment by AntonAL** [comment]
Tried to export on my machine, — no comments are presented in exported file.

File: apps/condo/domains/ticket/access/ExportTicketsService.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
-async function canExportTicketsToExcel ({ authentication: { item: user } }) {
-    if (!user) return false
-    if (user.isAdmin) return true
+const { throwAuthenticationError } = require('@condo/domains/common/utils/apolloErrorFormatter')
+const { checkOrganizationPermission } = require('@condo/domains/organization/utils/accessSchema')
+const get = require('lodash/get')
 
-    return { organization: { employees_some: { user: { id: user.id }, isBlocked: false } } }
+async function canExportTicketsToExcel ({ args: { data: { where } }, authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
```
Comment (pahaz) [inline apps/condo/domains/ticket/access/ExportTicketsService.js:6]: 🥇

File: apps/condo/pages/ticket/index.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const TicketsPage: IPageWithHeaderAction = () => {
     const CreateTicket = intl.formatMessage({ id: 'CreateTicket' })
     const EmergencyLabel = intl.formatMessage({ id: 'Emergency' })
     const DownloadExcelLabel = intl.formatMessage({ id: 'pages.condo.ticket.id.DownloadExcelLabel' })
-
+    const timeZone = intl.formatters.getDateTimeFormat().resolvedOptions().timeZone
```
Comment (Dimitreee) [inline apps/condo/pages/ticket/index.tsx:52]: It's better to use locale from organization

File: apps/condo/domains/ticket/schema/ExportTicketsService.js [hunk 3]
```diff
@@ -1,40 +1,33 @@
...
-                const { where, sortBy } = args.data
-                const organizationId = get(where, 'organization.id')
-                if (!organizationId) {
-                    throw new Error('[error] no organization id is passed')
-                }
-                const hasAccess = await checkOrganizationPermission(context.authedItem.id, organizationId, 'canManageTickets')
-                if (!hasAccess) {
-                    throw new Error('[error] you do not have access to this organization')
-                }
+                const { where, sortBy, timeZone } = args.data
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/ExportTicketsService.js:29]: is it possible to check timeZone argument?

File: apps/condo/domains/ticket/schema/ExportTicketsService.js [hunk 4]
```diff
@@ -44,7 +37,31 @@ const ExportTicketsService = new GQLCustomSchema('ExportTicketsService', {
...
+                const comments = await TicketComment.getAll(context, { ticket: { id_in: ticketIds } })
+                const indexedComments = {}
+                comments.forEach(comment => {
+                    if (!has(indexedComments, comment.ticket.id)) {
+                        indexedComments[comment.ticket.id] = []
+                    }
+                    indexedComments[comment.ticket.id].push(comment.content)
+                })
+                const statuses = await TicketStatus.getAll(context, { type_in: ['processing', 'canceled', 'completed'] })
+                const statusesIndexed = Object.fromEntries(statuses.map(({ id, type }) => ([id, type])))
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/schema/ExportTicketsService.js:53]:
```suggestion
                const indexeStatues = Object.fromEntries(statuses.map(({ id, type }) => ([id, type])))
```

**Review by Dimitreee — APPROVED** [review]

**Review by pahaz — APPROVED** [review]

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/ticket/schema/ExportTicketsService.js (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/schema/ExportTicketsService.js:53]: Done

File: apps/condo/domains/ticket/schema/ExportTicketsService.js (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/schema/ExportTicketsService.js:29]: Added DEFAULT_ORGANIZATION_TIMEZONE to organization common constants
Added "normalizeTimeZone" to timezone.js to common utils with spec. Works the same as "normalizePhone"
if passed timeZone is not valid will be used DEFAULT_ORGANIZATION_TIMEZONE

File: apps/condo/pages/ticket/index.tsx (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/pages/ticket/index.tsx:52]: Added task SBERDOMA-770

**Pull request merged**